tests/unit_tests/test_unit.py

```
@@ -2594,3 +2594,124 @@ def dummy(*args, **kwargs):
+def test_mgf_scan_index_built_correctly(mgf_small_with_scans, tmp_path):
+    """
+    _build_mgf_scan_index must return a dict mapping spectrum index
+    strings to the SCANS field values written into the MGF file.
+    """
+    from casanovo.data.ms_io import _build_mgf_scan_index
+
+    result = _build_mgf_scan_index(str(mgf_small_with_scans))
+    assert result == {"0": "17", "1": "42"}
+
+
+def test_mgf_without_scans_gives_empty_index(mgf_small, tmp_path):
```

Ruff (0.15.6)

[warning] Unused function argument: `tmp_path`

(ARG001)